odml/property.py

```diff
@@ -172,18 +202,41 @@ def _validate_values(self, values):
                 return False
         return True
 
-    @value.setter
-    def value(self, new_value):
-        # Make sure boolean value 'False' gets through as well...
-        if new_value is None or new_value == "":
-            return
+    def _convert_value_input(self, new_value):
+        """
+        This method ensures, that the passed new value is a list.
+        If new_value is a string, it will convert it to a list of
+        strings if the new_value contains embracing brackets.
+
+        returns list of new_value
+        """
         if isinstance(new_value, str):
             if new_value[0] == "[" and new_value[-1] == "]":
                 new_value = new_value[1:-1].split(",")
         if not isinstance(new_value, list):
             new_value = [new_value]
+        return new_value
+
+    @value.setter
+    def value(self, new_value):
+        """
+
+        Set the value of the property discarding any previous information.
+        Method will try to convert the passed value to the dtype of
+        the property and raise an ValueError, if not possible
+
+        :param new_value a single value or list of values.
+        """
+        # Make sure boolean value 'False' gets through as well...
+        if new_value is None or (isinstance(new_value, (list, tuple, str)) and len(new_value) == 0):
+            self._value = []
```

Comment:
I think Property.dtype is currently not reset when resetting the values. Maybe this can be reset here as well.

Reply:
we do have the dtype setter for this. Imho, I would rather not reset it on deleting the value(s)